packages/beacon-state-transition/src/allForks/util/cachedBeaconState.ts

```diff
@@ -192,6 +193,28 @@ export class BeaconStateContext<T extends allForks.BeaconState> {
     this.balances.persistent.asPersistent();
     this.previousEpochParticipation.persistent.asPersistent();
   }
+
+  /**
+   * Get domain from cache.
+   * domainType should always come from the config.
+   */
+  getDomain(domainType: DomainType, slot: phase0.Slot | null = null): Buffer {
+    const state = this.type.createTreeBacked(this.tree);
+    slot = slot || state.slot;
+    const forkName = this.config.getForkName(slot);
+    let domainByType = domainCache.get(forkName);
+    if (!domainByType) {
+      domainByType = new Map<DomainType, Buffer>();
+      domainCache.set(forkName, domainByType);
+    }
+    let domain = domainByType.get(domainType);
+    if (!domain) {
+      const forkVersion = this.config.getForkVersion(slot);
+      domain = computeDomain(this.config, domainType, forkVersion, state.genesisValidatorsRoot);
+      domainByType.set(domainType, domain);
+    }
+    return domain;
+  }
```

Comment:
Maybe move this method to EpochContext? This class (intentionally) doesn't have any application logic but rather defers to the various underlying caches.

In that case, `slot` can be a required param.

This shouldn't effect downstream consumers of the method since CachedBeaconState implements EpochContext. eg: usage will still be `state.getDomain(...)`

Also, the EpochContext can cache the genesisValidatorsRoot. (we could begin to consilidate the domain, fork digest, etc caching that requires config + genesisValidatorsRoot)